lib/ldapUser.class.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+include sfConfig::get('sf_root_dir') .'/vendor/password_compat/password.php';
```

Comment:
Perhaps sounds too complicated and not worth the time?

On Mon, May 2, 2016 at 3:05 PM, Mike Cantelon <notifications@github.com>
wrote:

> In lib/ldapUser.class.php
> <https://github.com/artefactual/atom/pull/328#discussion_r61813076>:
>
> > + *
> > + * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
> > + * it under the terms of the GNU Affero General Public License as published by
> > + * the Free Software Foundation, either version 3 of the License, or
> > + * (at your option) any later version.
> > + *
> > + * Access to Memory (AtoM) is distributed in the hope that it will be useful,
> > + * but WITHOUT ANY WARRANTY; without even the implied warranty of
> > + * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
> > + * GNU General Public License for more details.
> > + *
> > + * You should have received a copy of the GNU General Public License
> > + * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
> > + */
> > +
> > +include sfConfig::get('sf_root_dir') .'/vendor/password_compat/password.php';
>
> It looks like autoload will only handle classes, unfortunately. It parses
> files to determine if they're a class. We could subclass the autoloader
> class, though, as per:
>
> https://capocasa.net/trait-with-symfony-1-4
>
> I think the thing to do would be if a file, rather than a directory, is
> specified in autoload.yml then include it without parsing it.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/artefactual/atom/pull/328/files/a0611c3b67f116b00509c4db63ed4b49b7bc0b03#r61813076>
>



-- 
Jesús García Crespo,
Software Engineer, Artefactual Systems Inc.
http://www.artefactual.com | +1.604.527.2056